android/ui/src/main/java/com/trafi/ui/Badge.kt

```diff
@@ -0,0 +1,258 @@
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+import com.trafi.ui.BadgeType.MEDIUM_BADGE
+import com.trafi.ui.BadgeType.SMALL_BADGE
+import com.trafi.ui.component.internal.BadgeConstants
+import com.trafi.ui.theme.currentTheme
+
+@Composable
+private val constants
+    get() = BadgeConstants(currentTheme)
+
+data class BadgeInfo(
+    val text: String?,
+    val color: Color,
+    val textColor: Color? = null
+)
+
+@OptIn(ExperimentalLayout::class)
+@Composable
+fun Badge(
+    modifier: Modifier = Modifier,
+    badge: BadgeInfo?,
+    alternativeBadges: List<BadgeInfo> = listOf(),
+    vector: ImageVector? = null,
+    subbadge: ImageVector? = null,
+    badgeType: BadgeType,
+    isDisabled: Boolean = false
+) {
+    if (alternativeBadges.isEmpty()) {
+        SingleBadge(
+            badge = badge,
+            badgeType = badgeType,
+            modifier = modifier,
+            vector = vector,
+            subbadge = subbadge,
+            isDisabled = isDisabled
+        )
+    } else {
+        StackedBadge(
+            modifier = modifier,
+            badge = badge,
+            alternativeBadges = alternativeBadges,
+            vector = vector,
+            badgeType = badgeType
+        )
+    }
+
+}
+
+@OptIn(ExperimentalLayout::class)
+@Composable
+private fun SingleBadge(
+    modifier: Modifier = Modifier,
+    badge: BadgeInfo?,
+    vector: ImageVector? = null,
+    subbadge: ImageVector? = null,
+    badgeType: BadgeType,
+    isDisabled: Boolean = false
+) {
+    ConstraintLayout {
+        Surface(
+            color = if (isDisabled) constants.disabledColor else badge?.color ?: Color.Black,
+            contentColor = badge?.textColor ?: constants.defaultContentColor,
+            shape = RoundedCornerShape(getBadgeRounding(badgeType)),
+            modifier = modifier.sizeIn(minHeight = getBadgeHeight(badgeType))
+                .constrainAs(createRef()) {
+                    bottom.linkTo(parent.bottom)
+                }
+        ) {
+            BadgeFiller(
+                badgeType = badgeType,
+                vector = vector,
+                text = badge?.text,
+                isStackedBadge = false
+            )
+        }
+        subbadge?.let {
+            Image(
+                subbadge,
+                modifier = Modifier
+                    .width(constants.subBadgeIconWidth)
+                    .height(constants.subBadgeIconHeight)
+                    .constrainAs(createRef()) {
+                        bottom.linkTo(parent.bottom)
+                        top.linkTo(parent.bottom)
+                        start.linkTo(parent.end)
+                        end.linkTo(parent.end)
+                    }
+            )
+        }
+    }
+}
+
+
+@OptIn(ExperimentalLayout::class)
+@Composable
+private fun StackedBadge(
+    modifier: Modifier = Modifier,
```

Comment:
Compose convention is that `Modifier` is the first optional parameter (after all required parameters)